portfolio/src/main/webapp/script.js

```diff
@@ -178,71 +179,101 @@ function getIcon(sentimentScore) {
 }
 
 /**
- * Create element that represents a comment.
+ * Create element that represents a comment. Show option to delete comment
+ * if the logged in user made the comment.
  */
-function getCommentHtml(comment) {
+function getCommentHtml(comment, userId) {
   const tooltipUrl = 'https://cloud.google.com/natural-language/docs/basics' +
-      '#interpreting_sentiment_analysis_values';
+    '#interpreting_sentiment_analysis_values';
+  
+  /* Show delete button if it's the user's comment. */
+  let deleteButton = '';
+  if (comment.userId === userId) {
+    deleteButton = 
+        `<button class="delete-btn" onClick="deleteComment(${comment.id})">
+          <span class="material-icons">delete</span>
+        </button>`;
+  }
+
+  const sentimentIconTitle = 'Sentiment score: ' + 
+      comment.sentimentScore.toFixed(2) + '. Click to learn more.';
 
   const html =
       `<li>
-        <div class="comment">
-          <div>
-            <div class="comment-name">${comment.name}</div>
-            <div class="comment-text">${comment.text}</div>
-            <div class="comment-time">${parseTimestamp(comment.timestamp)}</div>
-          </div>
-          <div class="sentiment-icon">
-            <span class="material-icons"
-            title="Sentiment score: ${comment.sentimentScore.toFixed(2)}">
-            ${getIcon(comment.sentimentScore)}
-            </span>
-            <a href="${tooltipUrl}" target="_blank" class="more-info">?</a>
+        <div class="comment-content-wrapper">
+          <div class="comment">
+            <div class="comment-content">
+              <div class="comment-name">${comment.name}</div>
+              <div class="comment-text">${comment.text}</div>
+              <div class="comment-time">${parseTimestamp(comment.timestamp)}</div>
+            </div>
+            <div class="sentiment-icon">
+              <a href="${tooltipUrl}" target="_blank">
+                <span class="material-icons"
+                title="${sentimentIconTitle}">
+                ${getIcon(comment.sentimentScore)}
+                </span>
+              </a>
+            </div>
           </div>
         </div>
+        <div class="delete-btn-wrapper">
+          ${deleteButton}
+        </div>
       </li>`;
 
   return html;
 }
 
 /**
- * Delete all comments from datastore.
+ * Delete comment with id from datastore
  */
-function deleteAllComments() {
-  if (confirm('Are you sure you want to delete all comments, ' +
-    'including those from other users?')) {
-    fetch('/delete-data', {
+function deleteComment(id) {
+  if (confirm('Are you sure you want to permanently delete this comment?')) {
+    fetch(`/delete-comment?id=${id}`, {
       method: 'POST',
-    }).then(getComments());
+    }).then(() => {
+      /* Wait before refreshing to make
+       * sure datastore has been updated*/
+      setTimeout(function(){ location.reload(); }, 200);
```

Comment:
Is there a way to await datastore's update before refreshing, instead of using a magic number? (I don't know if there is, but if you haven't checked, maybe something to look into.)

Reply:
Why do you need the timeout? Shouldn't reloading the comments after deletion immediately remove the comment?